src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -275,23 +305,23 @@ public bool TryGet(ref Position position, out ReadOnlyMemory<byte> item, bool ad
             {
                 if (position == Position.First)
                 {
-                    item = _first;
-                    if (advance) { position.IntegerPosition++; position.ObjectPosition = _rest; }
+                    item = _data;
+                    if (advance) { position.IntegerPosition++; position.ObjectPosition = _next; }
                     return true;
                 }
                 else if (position.ObjectPosition == null) { item = default; return false; }
 
                 var sequence = (BufferListNode)position.ObjectPosition;
-                item = sequence._first;
+                item = sequence._data;
                 if (advance)
                 {
                     if (position == Position.First)
```

Comment:
What is the point to check `position` here? It's already compared to `Position.First` in the 306 line.